src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+    if (element === this.activeElement) return;
+    this.activeDescendant = element?.id;
+  }
+
+  get activeIndex() {
+    const idx = this.listItemElements.indexOf(this.activeElement);
+    return idx <= 0 ? idx : undefined;
+  }
+
+  set activeIndex(value) {
+    if (value === this.activeIndex) return;
+    if (!(isNil(value) || typeof value == 'number')) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be a number, null, or undefined`
+      );
+      return;
+    }
+    const { listItemElements } = this;
+    if (
+      typeof value === 'number' &&
+      (value < 0 || value > listItemElements.length - 1)
+    ) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be within the index range of data (0-${
+          listItemElements.length - 1
+        })`
+      );
+      return;
+    }
+    this.activeElement = listItemElements[value];
+  }
+
+  get selectedElement() {
+    return this.listboxElement.querySelector('[aria-selected]') ?? undefined;
+  }
+
+  set selectedElement(element) {
+    if (element === this.selectedElement) return;
+    this.selectedElement?.removeAttribute('aria-selected');
+    element?.setAttribute('aria-selected', 'true');
+  }
+
+  get selectedValue() {
+    return this.selectedElement?.id ?? undefined;
+  }
+
+  set selectedValue(value) {
+    if (value === this.selectedValue) return;
+    this.selectedElement = this.shadowRoot.getElementById(value);
+  }
+
+  render() {
+    const { selectedValue } = this;
+    const listItemEls = Array.from(this.data).map((value) => {
+      const listItemEl = document.createElement(this.itemRenderer);
+      listItemEl.setAttribute('role', 'option');
+      listItemEl.id = value;
+      listItemEl.innerText = this.format(value);
+      if (value === selectedValue) {
+        this.selectedElement = listItemEl;
+      }
+      return listItemEl;
+    });
+    this.listboxElement.replaceChildren(...listItemEls);
+  }
+
+  /**
+   * @description
+   *  Register events for the listbox interactions
+   */
+  registerEvents() {
+    this.addEventListener('focus', this.handleFocus);
+    this.addEventListener('keydown', this.handleKeyPress);
+    this.listboxElement.addEventListener('click', this.handleItemClick);
+  }
+
+  handleFocus(evt) {
+    // this.listboxElement.focus();
+    this.activeElement = this.selectedElement ?? this.listItemElements[0];
+  }
+
+  /**
+   * @description
+   *  Check if an item is clicked on. If so, set as both activeElement and selectedElement.
+   * @param evt
+   *  The click event object
+   */
+  handleItemClick(evt) {
+    if (evt.target.getAttribute('role') !== 'option') return;
+    this.selectedElement = evt.target;
+    this.activeElement = evt.target;
+  }
+
+  /**
+   * Check if the selected option is in view, and scroll if not
+   */
+  updateScroll() {
```

Comment:
this is to automatically scroll so the activeElement if the list items don't fit.

Reply:
nope! This was derived from the WAI design example, but I'd much rather use this. Will refactor and test. 🎉